eclair-core/src/test/scala/fr/acinq/eclair/io/MessageRelaySpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.typed.scaladsl.adapter.{ClassicActorRefOps, ClassicActorSystemOps}
+import akka.actor.{ActorContext, ActorRef}
+import akka.testkit.TestProbe
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.TestConstants.{Alice, Bob}
+import fr.acinq.eclair.io.Peer.ChannelFactory
+import fr.acinq.eclair.message.OnionMessages
+import fr.acinq.eclair.message.OnionMessages.{IntermediateNode, Recipient}
+import fr.acinq.eclair.wire.protocol.OnionMessage
+import fr.acinq.eclair.{TestKitBaseClass, randomKey}
+import org.scalatest.funsuite.AnyFunSuiteLike
+
+case class FakeChannelFactory(channel: TestProbe) extends ChannelFactory {
+  override def spawn(context: ActorContext, remoteNodeId: PublicKey, origin_opt: Option[ActorRef]): ActorRef = {
+    channel.ref
+  }
+}
+
+class MessageRelaySpec extends TestKitBaseClass with AnyFunSuiteLike {
```

Comment:
This test class should use the typed actor test framework.
I've done that change in https://github.com/ACINQ/eclair/commit/ece7bc08504fb5aba991e72d4f3ea93b1928ac6e and beefed up the integration tests.

It took me a lot more time than I expected, because I wanted to test that disconnected nodes that knew each other's addresses from previous interactions would automatically reconnect on-the-fly, but it was made difficult because we aggressively prune our DBs (which is good):

- when we don't have any channels with a peer we remove them from the `PeersDb`
- when a node doesn't have any visible channel on the network we remove them from the `NetworkDb`

So the test setup is a bit more complex now, but it lets us test more cases that match real-world usage.

Reply:
Done, I've merged your commit.